File: Cursuri/Grupe/Automation10/Automation10_Projects/Python_Automation/P2_Introduction_to_Selenium/exercitii_aditionale/date_picker.py
# avem nevoie sa importam cateva librarii gratuite care ne ajuta sa controlam chrome
from time import sleep
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.by import By

# initializam chrome
s = Service(ChromeDriverManager().install())
chrome = webdriver.Chrome(service=s)

# maximizam fereastra
chrome.maximize_window()

chrome.get("https://formy-project.herokuapp.com/form")

sleep(1)

date_picker = chrome.find_element(By.ID,"datepicker")
date_picker.click()
dates_list = chrome.find_elements(By.XPATH,"//table/tbody/tr/td[@class='day'] | //table/tbody/tr/td[@class='today day']")

date_evaluated = chrome.find_element(By.XPATH,"/html/body/div[2]/div[1]/table/thead/tr[2]/th[2]").text

print(f"We are going to extract the day for the 3d of {date_evaluated}")
for i in range(len(dates_list)):
      if dates_list[i].text=='3':
            dates_list[i].click()
            break

print(f"Date entered is: 3rd of {date_evaluated}")

sleep(6)

File: Personal/Java/Selenium/SeleniumWebDriver/Selenium WebDriver/Selenium_WebDriver/src/test/java/DatePicker.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;



public class DatePicker {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.chromedriver().setup();
        WebDriver driver = new ChromeDriver();
        driver.get(" https://formy-project.herokuapp.com/datepicker");
        WebElement dateField = driver.findElement(By.id("datepicker"));
        Thread.sleep(3000);
        dateField.sendKeys("22/08/2021");
        Thread.sleep(3000);
        dateField.sendKeys(Keys.RETURN);
        Thread.sleep(3000);
        driver.quit();

    }
}
